.github/workflows/generate-gql-types.yml

```diff
@@ -0,0 +1,58 @@
+name: Generate and Commit GQL Types
+
+on:
+  schedule:
+    # Run the workflow every Monday at midnight (UTC)
+    - cron: '0 0 * * MON'
+
+permissions:
+  contents: write  # Allows for committing and creating pull requests
+  actions: write
+
+jobs:
+  generate-gql-types:
+    name: Generating GQL Types 🦾
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [21.3.0]
+
+    steps:
+      # Checkout the repository
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Install Yarn/Node ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'yarn'
+      - name: Install Dependencies
+        run: yarn install --immutable
+      - name: Generate GQL Types
+        run: yarn generate
+      - name: Check for changes
+        run: git diff --exit-code -- typedefs/
+        continue-on-error: true
```

Comment:
Why do we want to continue on error here? 🤔

Reply:
@theyokohamalife Based on my understanding of how these work. The `git diff --exit-code` would return a non zero code and therefore return an error. Letting it not proceed even though we want the rest of the workflow to run in opening a branch and committing the changes.